src/python/zquantum/core/cost_function.py

```diff
@@ -111,19 +116,21 @@ def __init__(
         target_operator: SymbolicOperator,
         ansatz: Dict,
         backend: QuantumBackend,
+        estimator: Estimator = BasicEstimator(),
```

Comment:
Generally we don't want default values to be mutable types because it can lead to unexpected behavior. You can instead have the default value be `None` and have an if statement in the initializer:
```python
if estimator is None:
  estimator = BasicEstimator()
```